HLTrigger/HLTcore/interface/TriggerExpressionOperators.h

```diff
@@ -107,16 +117,44 @@ namespace triggerExpression {
     OperatorXor(Evaluator* arg1, Evaluator* arg2) : BinaryOperator(arg1, arg2) {}
 
     bool operator()(const Data& data) const override {
-      // force the execution af both arguments, otherwise precalers won't work properly
+      // force the execution of both arguments, otherwise prescalers won't work properly
       bool r1 = (*m_arg1)(data);
       bool r2 = (*m_arg2)(data);
       return r1 xor r2;
     }
 
     void dump(std::ostream& out) const override {
+      out << '(';
       m_arg1->dump(out);
       out << " XOR ";
       m_arg2->dump(out);
+      out << ')';
+    }
+  };
+
+  class OperatorMasking : public BinaryOperator {
+  public:
+    OperatorMasking(Evaluator* arg1, Evaluator* arg2) : BinaryOperator(arg1, arg2) {}
+
+    bool operator()(const Data& data) const override {
+      // force the execution of both arguments, otherwise prescalers won't work properly
+      bool r1 = (*m_arg1)(data);
+      (*m_arg2)(data);
```

Comment:
IS this needed, if `MASKING` does not support prescaled right-hand arguments ?

Reply:
I would say no, indeed, but I wasn't 100% sure about it because of the comment on the prescalers. Since a prescale operator cannot be on the right side of `MASKING` in any case, I think it's safe to remove this.